### PR TITLE
Vlapic: refine lapic for MISRA-C

### PR DESCRIPTION
**Pull request by lifeix**
v3:
leave theoretic lapic hardware issue to machine check

v1-v2:
addrsss some comments about MASRA-C voilation and coding style.

v1:
This serial in order to eliminate the MISRA-C violations in lapic.
And fix a minor bug when resume from s3.

Tracked-On: #861

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
1869<br>No new violations to the coding guideline detected.

File: hypervisor/arch/x86/lapic.c [hunk 1]
```diff
@@ -45,20 +45,23 @@ static union lapic_base_msr saved_lapic_base_msr;
...
 	 * does not get hung up due to already set "in-service" interrupts left
 	 * over from the boot loader environment. This actually occurs in real
 	 * life, therefore we will ensure all the in-service bits are clear.
 	 */
-	do {
-		if (msr_read(isr_reg) != 0U) {
-			msr_write(MSR_IA32_EXT_APIC_EOI, 0U);
-			continue;
+	for (isr_reg = MSR_IA32_EXT_APIC_ISR7; isr_reg >= MSR_IA32_EXT_APIC_ISR0; isr_reg--) {
+		for (acked = 0U; acked < 32U; acked++) {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/lapic.c]: don't use acked, use i/j/k is better.

File: hypervisor/arch/x86/lapic.c [hunk 2]
```diff
@@ -45,20 +45,23 @@ static union lapic_base_msr saved_lapic_base_msr;
...
+			if (msr_read(isr_reg) != 0U) {
+				msr_write(MSR_IA32_EXT_APIC_EOI, 0U);
+			} else {
```
Comment (dongyaozu) [inline hypervisor/arch/x86/lapic.c]: Here I prefer you add a variable  isr_val, and: isr_val =msr_read(isr_reg).
This way, you can remove break, and do condition check in for loop.

File: hypervisor/arch/x86/lapic.c (hunk 1 above)
Comment (lifeix) [reply hypervisor/arch/x86/lapic.c]: done

**Comment by HuiHuangShi** [comment]
1869<br>No new violations to the coding guideline detected.

**Pull request merged**